pints/_log_likelihoods.py

```diff
@@ -133,6 +133,131 @@ def __call__(self, x):
+    have `n_outputs` fewer parameter dimensions.
+    The log-likelihood is given in terms of the sum of squared errors:
+    .. math::
+        SSE = \sum_{i=1}^n (f_i(\\theta) - y_i)^2
+    and is given up to a normalisation constant by:
+    .. math::
+        \text{log } L = - n / 2 \text{log}(\pi) -
+            \text{log}(2 (b - a) \sqrt(2)) +
+            (1 / 2 - n / 2) \text{log}(SSE) +
+            \text{log}\\left[\Gamma((n - 1) / 2, \frac{SSE}{2 b^2}) -
+            \Gamma((n - 1) / 2, \frac{SSE}{2 a^2}) \right]
+    where :math:`\Gamma(u,v)` is the upper incomplete gamma function
+    as defined here: https://en.wikipedia.org/wiki/Incomplete_gamma_function
+    This log-likelihood is inherently a Bayesian method since it
+    assumes a uniform prior on :math:`\sigma\sim U(a,b)`. However
+    using this likelihood in optimisation routines should yield the
+    same estimates as the full `pints.GaussianLogLikelihood`.
+    Arguments:
+    ``problem``
+        A :class:`SingleOutputProblem` or :class:`MultiOutputProblem`.
+    ``lower``
+        The lower limit on the uniform prior om `sigma`. Must be
+        non-negative.
+    ``upper``
+        The upper limit on the uniform prior om `sigma`.
+    *Extends:* :class:`ProblemLogLikelihood`
+    """
+
+    def __init__(self, problem, lower, upper):
+        super(GaussianIntegratedUniformLogLikelihood, self).__init__(problem)
+
+        # Get number of times, number of outputs
+        self._nt = len(self._times)
+        self._no = problem.n_outputs()
+
+        # Add parameters to problem
+        self._n_parameters = problem.n_parameters()
+        a = lower
+        if np.isscalar(a):
+            a = np.ones(self._no) * float(a)
+        else:
+            a = pints.vector(a)
+            if len(a) != self._no:
+                raise ValueError(
+                    'Lower limit on uniform prior for sigma must be a ' +
+                    ' scalar or a vector of length n_outputs.')
+        if np.any(a < 0):
+            raise ValueError('Lower limit on uniform prior for sigma ' +
+                             'must be non-negative.')
+        b = upper
+        if np.isscalar(b):
+            b = np.ones(self._no) * float(b)
+        else:
+            b = pints.vector(b)
+            if len(b) != self._no:
+                raise ValueError(
+                    'Upper limit on uniform prior for sigma must be a ' +
+                    ' scalar or a vector of length n_outputs.')
+        if np.any(b <= 0):
+            raise ValueError('Lower limit on uniform prior for sigma ' +
+                             'must be positive.')
+        diff = b - a
+        if np.any(diff <= 0):
+            raise ValueError('Upper limit on uniform prior for sigma ' +
+                             'must exceed lower limit.')
+        self._a = a
+        self._b = b
+
+        # Pre-calculate
+        n = self._nt
+        self._n_minus_1_over_2 = (n - 1.0) / 2.0
```

Comment:
Don't need to change this, but the .0 business is no longer required in python 3. We add a "from future import division" at the top of every file to get the same behaviour on python 2. So throughout pints you can just write `1 / 2` and the result will be `0.5`

Reply:
Ok, I'm just paranoid, haha!